com.microsoft.mrtk.accessibility/Subsystems/MRTK/MRTKAccessibilitySubsystem.cs

```diff
@@ -44,6 +45,142 @@ public AccessibilityProvider() : base()
+                    return false;
+                }
+
+                if (!describableObjects[classification].Remove(describableObject))
+                {
+                    Debug.LogError($"Failed to unregister {describableObject.name} as a describable object of classification {classification}");
+                    return false;
+                }
+
+                return true;
+            }
+
+            /// <summary>
+            /// Collects the registered <see cref="GameObject"/>s from the requested <see cref="DescribableObjectClassification"/>s.
+            /// </summary>
+            /// <param name="classifications">The combined flags specifiying the desired classification(s) (people, places, things, etc.).</param>
+            /// <param name="describableObjectsList">The collection which will receive the requested <see cref="GameObject"/>s.</param>
+            /// <remarks>When this method is called, the objectList will be cleared prior to adding the requested <see cref="GameObject"/>s.</remarks>
+            private void AssembleDescribableObjects(
+                List<DescribableObjectClassification> classifications,
+                List<GameObject> describableObjectsList)
+            {
+                foreach (DescribableObjectClassification classification in classifications)
+                {
+                    describableObjectsList.AddRange(describableObjects[classification]);
+                }
+            }
+
+            /// <summary>
+            /// Filters the provided collection of <see cref="GameObject"/>s based on the visibility to the main camera
+            /// and the maximum allowed distance.
+            /// </summary>
+            /// <param name="visibility">The desired visibility (in field of view, full surround, etc.) of the <see cref="GameObject"/>s.</param>
+            /// <param name="maxDistance">The distance, in meters, beyond which <see cref="GameObject"/>s will be filtered.</param>
+            /// <param name="objectList">The collection of <see cref="GameObject"/>s that will be filtered based on visibility and distance.</param>
+            private void FilterDescribableObjects(
+                DescribableObjectVisibility visibility,
+                float maxDistance,
+                List<GameObject> objectList)
+            {
+                // Ensure there is a collection to filter.
+                if (objectList.Count == 0) { return; }
+
+                float maxDistanceSquared = maxDistance * maxDistance;
+
+                // Walk the list backwards, so that it does not hit a collection changed exception.
+                for (int i = (objectList.Count - 1); i != 0; i--)
+                {
+                    GameObject obj = objectList[i];
+
+                    // Does the object have a collider?
+                    Collider collider = obj.GetComponent<Collider>();
```

Comment:
nit: to save on allocations in-editor for non-existent colliders, could this be `if (!obj.TryGetComponent(out Collider collider))`?

Reply:
100%

Reply:
done 😺